### PR TITLE
Changed the sh2 and sh4 disassemblers to use 'std::ostream &' internally

### PR DESCRIPTION
**Pull request by ajrhacker**

### CODE REVIEW

File: src/devices/cpu/sh2/sh2dasm.cpp [hunk 1]
```diff
@@ -15,596 +15,600 @@ static const char *const regname[16] = {
...
+		case  6: flags = op0110(stream, pc, opcode); break;
+		case  7: flags = op0111(stream, pc, opcode); break;
+		case  8: flags = op1000(stream, pc, opcode); break;
+		case  9: flags = op1001(stream, pc, opcode); break;
+		case 10: flags = op1010(stream, pc, opcode); break;
+		case 11: flags = op1011(stream, pc, opcode); break;
+		case 12: flags = op1100(stream, pc, opcode); break;
+		case 13: flags = op1101(stream, pc, opcode); break;
+		case 14: flags = op1110(stream, pc, opcode); break;
+		default: flags = op1111(stream, pc, opcode); break;
```
Comment (cuavas) [inline src/devices/cpu/sh2/sh2dasm.cpp]: case labels don't introduce an additional scope level.  Please revert the indentation to how it was before.

Comment (ajrhacker) [inline src/devices/cpu/sh2/sh2dasm.cpp]: The MAME coding style guidelines don't actually state what indentation level case labels use, and there remains no general consensus on this point in the wider world of programming, let alone other extant parts of the MAME codebase. Nevertheless, I'll move the indentation back here.

Comment (cuavas) [inline src/devices/cpu/sh2/sh2dasm.cpp]: Yeah, but if you look at the rest of the file, it doesn't add the extra indentation for non-existent scope.  So it's bad on two fronts: you're making the file inconsistent with itself, and making the indentation not follow scoping (a label never creates an additional scope level).

**Comment by ajrhacker** [comment]
Rebased with the requested change incorporated.

File: src/devices/cpu/sh4/sh4dasm.cpp [hunk 2]
```diff
@@ -15,253 +15,279 @@ static const char *const regname[16] = {
...
-		case 0x40:
-			sprintf(buffer,"CLRS"); break;
-		case 0x50:
-			sprintf(buffer,"SETS"); break;
+			case 0x00: stream << "CLRT"; break;
+			case 0x10: stream << "SETT"; break;
+			case 0x20: stream << "CLRMAC"; break;
+			case 0x30: stream << "LDTLB"; break;
+			case 0x40: stream << "CLRS"; break;
+			case 0x50: stream << "SETS"; break;
```
Comment (cuavas) [inline src/devices/cpu/sh4/sh4dasm.cpp]: Here's a case where you've still made the file inconsistent with itself.

File: src/devices/cpu/sh4/sh4dasm.cpp [hunk 3]
```diff
@@ -15,253 +15,279 @@ static const char *const regname[16] = {
...
-			sprintf(buffer,"MOVT    %s", regname[Rn]); break;
+			case 0x00:
+				stream << "NOP";
+				break;
+			case 0x10:
+				stream << "DIV0U";
+				break;
+			case 0x20:
+				util::stream_format(stream, "MOVT    %s", regname[Rn]);
+				break;
```
Comment (cuavas) [inline src/devices/cpu/sh4/sh4dasm.cpp]: Here's another case.

**Comment by ajrhacker** [comment]
Rebased yet again. This time the spacing really should be consistent.

**Pull request merged**